brave/src/test/java/com/linecorp/armeria/server/brave/BraveServiceIntegrationTest.java

```diff
@@ -84,8 +86,16 @@ protected void init() {
         // "/nested" left out as there's no sub-routing feature at the moment.
 
         sb.service("/child", (ctx, req) -> {
+
+            // The `Span.start()` and `Span.finish()` estimate the current time using
+            // `baseEpochMicros + (System.nanoTime() - baseTickNanos)`.
+            // However, Armeria achieves the current time using `System.currentTimeMillis() * 1000` in Java8.
+            // The two different way of getting the current time could cause an error of a millisecond.
+            // Sometimes the tolerance makes the server appear to finish earlier than child in test case.
+            // This is not a problem in the Brave{Service,Client} because they only use Armeria timestamp.
             tracing.tracer().nextSpan().name("child").start().finish();
-            return HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "happy");
+            return HttpResponse.delayed(
+                    HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "happy"), Duration.ofMillis(1));
```

Comment:
How about delaying only for Java 8? We use microsecond precision for 9+.

Reply:
That's a good idea.